app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt

```diff
@@ -499,10 +517,69 @@ class DeviceShieldTrackerActivity :
         data class Denied(val intent: Intent) : VpnPermissionStatus()
     }
 
+    private fun showAppTpEnabledCta() {
+        supportFragmentManager.findFragmentByTag(TAG_APPTP_ENABLED_CTA_DIALOG)?.let {
+            return
+        }
+
+        val dialog = TypewriterDaxDialog.newInstance(
+            daxText = getString(R.string.atp_ActivityAppTpEnabledCtaText),
+            primaryButtonText = getString(R.string.atp_ActivityAppTpEnabledCtaButtonLabel),
+            hideButtonText = "",
+        )
+
+        dialog.setDaxDialogListener(object : DaxDialogListener {
+            override fun onDaxDialogDismiss() {
+                // NO OP
+            }
+
+            override fun onDaxDialogPrimaryCtaClick() {
+                dialog.dismiss()
+                launchKonfetti()
+                deviceShieldPixels.didPressOnAppTpEnabledCtaButton()
+            }
+
+            override fun onDaxDialogSecondaryCtaClick() {
+                // NO OP
+            }
+
+            override fun onDaxDialogHideClick() {
+                // NO OP
+            }
+        })
+
+        handler.postDelayed(delayInMillis = APPTP_ENABLED_CTA_DIALOG_DELAY_MS) {
```

Comment:
what's the reason for this?

Reply:
The handler will be removed in the last PR.